### PR TITLE
tests: improve tests for ide

### PR DESCRIPTION
**Pull request by Chris53897**
This helps the IDE (in my case PhpStorm) to read the code.
Please have a look if you find it useful and is correct.

### CODE REVIEW

File: tests/Functional/RepositoryProxyTest.php [hunk 1]
```diff
@@ -342,7 +344,9 @@ public function can_use_get_count(): void
         $this->assertSame(4, $categoryFactoryClass::repository()->getCount());
     }
 
+    /** @phpstan-return \Zenstruck\Foundry\Tests\Fixtures\Entity\Category|\Zenstruck\Foundry\Tests\Fixtures\Document\ODMCategory */
     abstract protected function categoryClass(): string;
 
+    /** @phpstan-return \Zenstruck\Foundry\Tests\Fixtures\Factories\CategoryFactory|\Zenstruck\Foundry\Tests\Fixtures\Factories\ODM\CategoryFactory */
     abstract protected function categoryFactoryClass(): string;
```
Comment (nikophil) [inline tests/Functional/RepositoryProxyTest.php:351]: - these methods does not return actual objects but their class, so we can use `class-string` 
- I think there is no need for the `phpstan-` prefix?

```suggestion
    /** @return class-string<ODMCategory|ORMCategory> */
    abstract protected function categoryClass(): string;
    
    /** @return class-string<ODMCategoryFactory|ORMCategoryFactory> */
    abstract protected function categoryFactoryClass(): string;
```

Comment (nikophil) [inline tests/Functional/RepositoryProxyTest.php]: - these methods does not return actual objects but their class, so we can use `class-string` 
- I think there is no need for the `phpstan-` prefix?

```suggestion
    /** @return class-string<ODMCategory|ORMCategory> */
    abstract protected function categoryClass(): string;
    
    /** @return class-string<ODMCategoryFactory|ORMCategoryFactory> */
    abstract protected function categoryFactoryClass(): string;
```

File: tests/Functional/RepositoryProxyTest.php [hunk 2]
```diff
@@ -93,6 +93,7 @@ public function assertions(): void
     /**
      * @test
      * @group legacy
+     * @noinspection PhpDeprecationInspection
```
Comment (nikophil) [inline tests/Functional/RepositoryProxyTest.php]: not sure we should add PHPStorm specific annotations? :thinking: 

I'm not even sure we should hide that these methods use deprecated methods...

File: tests/Functional/ProxyTest.php [hunk 3]
```diff
@@ -290,6 +290,7 @@ public function without_auto_refresh_keeps_disabled_if_originally_disabled(): vo
         $this->assertSame('another new body', $post->getBody());
     }
 
+    /** @phpstan-return \Zenstruck\Foundry\Tests\Fixtures\Factories\PostFactory|\Zenstruck\Foundry\Tests\Fixtures\Factories\ODM\PostFactory */
     abstract protected function postFactoryClass(): string;
```
Comment (nikophil) [inline tests/Functional/ProxyTest.php]:
```suggestion
    /** @return class-string<ODMPostFactory|ORMPostFactory> */
    abstract protected function postFactoryClass(): string;
```

File: src/Test/TestState.php [hunk 4]
```diff
@@ -78,6 +78,7 @@ public static function disableDefaultProxyAutoRefresh(): void
 
     /**
      * @deprecated Use TestState::enableDefaultProxyAutoRefresh()
+     * @noinspection PhpDeprecationInspection
```
Comment (nikophil) [inline src/Test/TestState.php]: I don't really understand why we should hide that we're using a deprecated method?

**Review by nikophil — CHANGES_REQUESTED** [review]
hi @Chris53897 thanks for your PR

IMO IDE-specific annotations should not land in the code.

File: src/Test/TestState.php (hunk 4 above)
Comment (Chris53897) [reply src/Test/TestState.php]: I have reverted it

File: tests/Functional/RepositoryProxyTest.php (hunk 2 above)
Comment (Chris53897) [reply tests/Functional/RepositoryProxyTest.php]: I have reverted it

**Review by nikophil — APPROVED** [review]

**Comment by Chris53897** [comment]
I changed it and add the use statements.

PhpStorm does not yet understand it like this https://youtrack.jetbrains.com/issue/WI-55925.
I guess it can only understand it like this.
https://blog.jetbrains.com/phpstorm/2021/07/phpstorm-2021-2-release/#class-string

But maybe other ide's can understand it now. And hopefully phpstorm in the future

**Comment by nikophil** [comment]
> PhpStorm does not yet understand it

I've tried your PR, and trying to "ctrl + click" `createMany()` in this code, and it works pretty well 
```php
// \Zenstruck\Foundry\Tests\Functional\RepositoryProxyTest

    public function can_find_random_object(): void
    {
        $this->categoryFactoryClass()::createMany(5);
```

**Comment by nikophil** [comment]
if PHPStorm does not understand it, we could change like this, maybe?

```php

    /** @return class-string */
    abstract protected function categoryClass(): string;

    /** @return class-string<ModelFactory> */
    abstract protected function categoryFactoryClass(): string;
```

**Pull request closed without merging**